packages/react/src/SelectPanel/SelectPanel.tsx

```diff
@@ -459,6 +464,16 @@ export function SelectPanel({
           role: 'dialog',
           'aria-labelledby': titleId,
           'aria-describedby': subtitle ? subtitleId : undefined,
+          ...(variant === 'modal'
+            ? {
+                /* override AnchoredOverlay position */
```

Comment:
Note for reviewer: This is where all the shame lies. We are still using an AnchoredOverlay, but then we override it's position by directly passing top, left, etc to overlayProps going down

Reply:
fair

Reply:
I'd rather us have branching on the container component instead of doing this, as it semantically is much harder to implement. I think it's fine for now, but definitely worth improving.

Reply:
^ @hectahertz do we have an idea on the effort to do branching off the get go? I agree with you and we do have a bit of leeway time-wise with this, wondering if we can get it in 👀